subprojects/core/src/main/java/org/gradle/api/internal/project/DefaultProject.java

```diff
@@ -1096,34 +1111,41 @@ public Object configure(Object object, Closure configureClosure) {
         return objects;
     }
 
+    private <T> Iterable<T> configure(Iterable<T> objects, Closure configureClosure, String callerContextInformation) {
+        for (T object : objects) {
+            gradle.getDomainObjectConfigurator().configure(object, configureClosure, callerContextInformation);
```

Comment:
Instead of a new service I think we should have a `BuildOperationAction`, because we can pass that around. The service pattern would only work for callbacks that are executed immediately. In fact, when you want to execute immediately, it doesn't make much sense to have another indirection layer, you can just use the BuildOperationExecutor directly.